### PR TITLE
refactor: exporting nodebuilder/header.InitStore, moving mock to nodebuilder/header

### PR DESCRIPTION
**Pull request by distractedm1nd**
## Overview
Semi-old branch I forgot to open a PR for, that is related to #1458. Although we cannot fix the ugly error message without https://github.com/uber-go/fx/issues/999, we can get rid of an ugly todo in a start hook.

This PR is purely a suggestion, and if the team doesn't like it we can just close it. 

## Checklist

- [X] New and updated code has appropriate documentation
- [X] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [X] Linked issues closed with keywords

### CODE REVIEW

File: header/p2p/exchange_test.go [hunk 1]
```diff
@@ -12,12 +11,11 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
-	tmbytes "github.com/tendermint/tendermint/libs/bytes"
-
 	"github.com/celestiaorg/go-libp2p-messenger/serde"
 
 	"github.com/celestiaorg/celestia-node/header"
 	p2p_pb "github.com/celestiaorg/celestia-node/header/p2p/pb"
+	headerMock "github.com/celestiaorg/celestia-node/nodebuilder/header/mocks"
```
Comment (Wondertan) [inline header/p2p/exchange_test.go:18]: 😠

Comment (Wondertan) [reply header/p2p/exchange_test.go:18]: It's a custom mock, so we can keep it somewhere in to the header pkg

Comment (Wondertan) [reply header/p2p/exchange_test.go:18]: Nodebuilder can be used to store autogen mocks

**Review by Wondertan — CHANGES_REQUESTED** [review]
Very Nice! Suprised this idea didn't come when we introduced `initStore`. A bit concerned about it being public, but definitely worth it.

Requesting change to mock location

File: header/p2p/exchange_test.go (hunk 1 above)
Comment (distractedm1nd) [reply header/p2p/exchange_test.go]: thanks, fixed

**Review by renaynay — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Pull request merged**